### ISSUE TITLE
Investigate box-shadow and "prefers contrast"/"high contrast"

### ISSUE THREAD

**Issue by taysea** [grommet/hpe-design-system#4057]
Related to https://github.com/grommet/hpe-design-system/issues/3903

We are considering using box-shadow to style borders of components. That said, there is an issue with this approach for ["box-shadow" in Windows high contrast mode](https://github.com/grommet/grommet/issues/6155), where box-shadows are removed. This would result in all of our components having no borders in high-contrast.

Before we dive too deep into building with box-shadow for borders, let's better understand how box-shadow is handled in the following scenarios:
- [Windows high-contrast](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/)
- [CSS prefers-contrast](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast)
- [Mac high-contrast](https://support.apple.com/lv-lv/guide/mac-help/unac089/mac)?

We will use [Gitlab Pajamas buttons](https://design.gitlab.com/components/button) to test because they style their button borders using the box-shadow approach.

As a baseline, here is what the buttons look like:

![Image](https://github.com/user-attachments/assets/dd7fc9c6-acbc-4f14-b58b-7cd7282ef005)

**Deliverables:**

Windows high-contrast:
- [x] Screenshot of Gitlab buttons in "default" state
- [x] Screenshot of a button in "focus" state
- [x] Screenshot of CSS styles from browser inspect for both of the above

Mac Chrome:
- [x] Screenshot of Gitlab buttons in "default" state
- [x] Screenshot of a button in "focus" state
- [x] Screenshot of CSS styles from browser inspect for both of the above

Mac Safari **(no free options available for high contrast)**:
- [ ] Screenshot of Gitlab buttons in "default" state
- [ ] Screenshot of a button in "focus" state
- [ ] Screenshot of CSS styles from browser inspect for both of the above

Mac Firefox with preferences color override:
- [x] Screenshot of Gitlab buttons in "default" state
- [x] Screenshot of a button in "focus" state
- [x] Screenshot of CSS styles from browser inspect for both of the above

Mac high-contrast from system settings + Chrome:
- [x] Screenshot of Gitlab buttons in "default" state
- [x] Screenshot of a button in "focus" state
- [x] Screenshot of CSS styles from browser inspect for both of the above

Recommendation:
- [ ] What is the recommendation to continue exploring use of box-shadow for styling borders?

**Comment by jcfilben** [comment]
Mac Firefox with preferences colors override set to always

<img width="1218" alt="Screenshot 2024-07-29 at 12 46 06 PM" src="https://github.com/user-attachments/assets/cc921a50-20cb-45d9-a7b4-f0fdaad8e773">

<img width="1501" alt="Screenshot 2024-07-29 at 12 10 32 PM" src="https://github.com/user-attachments/assets/ce6e70f2-b4a8-43d3-9219-be04655254a2">

<img width="1484" alt="Screenshot 2024-07-29 at 12 10 53 PM" src="https://github.com/user-attachments/assets/30f4efda-821c-4861-8ed3-9d921a3b979f">

**Comment by jcfilben** [comment]
Mac high-contrast from system settings + Chrome
The high contrast mode in mac doesn't affect the web page

<img width="1792" alt="Screenshot 2024-07-29 at 12 54 28 PM" src="https://github.com/user-attachments/assets/00724651-25ef-4665-b494-1ca00b508b00">

With focus
<img width="1792" alt="Screenshot 2024-07-29 at 12 55 03 PM" src="https://github.com/user-attachments/assets/4e370cfb-0c36-4c07-80be-19cba2ea9827">

**Comment by jcfilben** [comment]
Mac + chrome high contrast plugin 
https://chromewebstore.google.com/detail/high-contrast/djcfdncoelnlbldjfhinnjlhdjlikmph?hl%3Den&pli=1

<img width="1791" alt="Screenshot 2024-07-29 at 1 00 24 PM" src="https://github.com/user-attachments/assets/7cf188d0-88f3-45cf-a1cb-6f4b0875b76a">

with focus
<img width="1792" alt="Screenshot 2024-07-29 at 1 00 58 PM" src="https://github.com/user-attachments/assets/fac2b785-8493-49ee-89e0-6054ff193af4">

**Comment by jcfilben** [comment]
Windows high contrast mode

box-shadow is not displayed in windows high contrast mode. Pajamas uses border instead of box-shadow for the border of the button and outline for the focus indicator.

![Screenshot 2024-07-29 154501](https://github.com/user-attachments/assets/5c02a508-9792-4bd5-b6ab-f731a25035ba)
![Screenshot 2024-07-29 154518](https://github.com/user-attachments/assets/82b456af-baa6-4972-a08e-238ded55d736)
![Screenshot 2024-07-29 154541](https://github.com/user-attachments/assets/9c85d469-2a2d-48fa-8a68-e125c7ce6efa)
![Screenshot 2024-07-29 154603](https://github.com/user-attachments/assets/75aee154-4203-4317-a36a-2c0656b75dc0)

**Comment by jcfilben** [comment]
Quick search on the number of people using windows high contrast mode
![Screenshot 2024-07-29 at 2 55 39 PM](https://github.com/user-attachments/assets/2011a53e-10fa-4848-b499-aa1d867722e8)


It looks like Pajamas is using `forced-colors` https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors to determine if `box-shadow` or `border` should be used.

If we want to move forward with replacing borders with `box-shadow` we could take a similar approach. Teams that are using tokens would need to manually add this check into their code to determine if the browser is in a forced colors mode and then use `border` instead of `box-shadow`.

**Comment by taysea** [comment]
Thank you for the work and shareout, closing this ticket as done.
